pkg/resources/statefulsets/statefulset.go

```diff
@@ -344,8 +344,15 @@ func NewForCluster(cluster *v1alpha1.Cluster, images operatoropts.Images, servic
 		})
 	}
 
+	var serverImage string
+	if cluster.Spec.Repository != "" {
```

Comment:
This defaulting will be redundant if we can manage to default in `Cluster.EnsureDefaults()`